homeassistant/components/modbus/modbus.py

```diff
@@ -2,11 +2,15 @@
 import logging
 import threading
 
-from pymodbus.client.sync import ModbusSerialClient, ModbusTcpClient, ModbusUdpClient
-from pymodbus.constants import Defaults
-from pymodbus.exceptions import ModbusException
-from pymodbus.pdu import ExceptionResponse, IllegalFunctionRequest
-from pymodbus.transaction import ModbusRtuFramer
+from pymodbus.client.sync import (  # type: ignore
```

Comment:
And are these needed?

Reply:
no, we should really warn people about how to run mypy. they are needed if you do:

cd homeassistant/components/modbus
Mypy .

but not if you do:
cd core
Mypy homeassistant/components/modbus

Corrects.

Reply:
The official way to run mypy locally is `tox -e typing` from the repo root. Although it's a bit slow for iterating. I find it easier to install `requirements_test_all.txt` in current venv and run `mypy homeassistant` directly.

Reply:
whether you run "tox -e typing" or "mypy ..." it is the same problem you need to be in root, and get no warning if you are not. I also run "pytest" but that runs best in the integrations (test path).

I now run mypy in root, but it something that should be documented, because it is not really what you expect.